src/Framework/LogInterpolatingHandler.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Build.Framework
+{
+    [InterpolatedStringHandler]
+    public ref struct LogInterpolatedStringHandler
+    {
+        private char[] buffer;
+        private int position = 0;
+        private int argPosition = 0;
+
+        public object[] Arguments { get; } = Array.Empty<object>();
+
+        public LogInterpolatedStringHandler(int literalLength, int formattedCount)
+        {
+            if (formattedCount > 99)
+            {
+                throw new ArgumentOutOfRangeException("Number of formatted arguments must be less than 100.");
+            }
+
+            // Length is computed with reserved space for "{x}" and "{xx}" placeholders 
+            buffer = new char[literalLength + (4 * formattedCount)];
```

Comment:
Alternatively we can allocate exact - most cases will have under 10 args anyways

```suggestion
            buffer = new char[literalLength + 3 * (formattedCount % 10) + 4 * (formattedCount - (formattedCount % 10))];
```
(removing brackets for multiplication to make it less bracket crazy)

Reply:
Yes, we can be more precise. I modified code based on your comment, but thinking about it, the array lifetime is very short and it will most likely not survive Gen0 collection anyway. Maybe we can use ```ArrayPool``` to remove allocations, but I'm not sure if the added runtime complexity outweighs memory saving. @ladipro ?

Reply:
Okay, thinking about it even more, using ```ArrayPool``` isn't in this case probably possible as we don't know when to return rented array back to the pool :)

Reply:
If we go this route of composing a string for `string.Format`, then I think it's fine to have a reasonable upper bound. Even `(formattedCount * (2 + numberOfDigitsInFormattedCount))` could be OK.

Reply:
I wanted to keep it simple for >10 args scenario, so I used ```formattedCount * (maxNumberOfDigits + 2)``` in new commit.